### DISCUSSION TITLE
Error: ReferenceError: window is not defined

### DISCUSSION THREAD

**Discussion by sdoigmm** [webdriverio/webdriverio#11562] [Technical Discussion]
My test has 

`await(await $('.upper-canvas')).moveTo(862,242); await(await $('.upper-canvas')).click();`

but this position is missed by about 200px to the left.

To debug this I created a file `test.js`:

```
class getCursorPosition {

    constructor() {
        window.addEventListener('load', canvasCursorPosition, false);

        function canvasCursorPosition() {
            canvas = document.querySelector('canvas.upper-canvas');
            canvas.addEventListener('mousedown', function(e) {
                getCursorPosition(canvas, e)
            })
        }
    }

    getCursorPosition(canvas, event) {
        const rect = canvas.getBoundingClientRect()
        const x = event.clientX - rect.left
        const y = event.clientY - rect.top
        console.log("x: " + x + " y: " + y)
    }

}

export default new getCursorPosition();
```

(I want to console.log the position clicked by `$('.upper-canvas')).click();`).

In my original test file I have `import getCursorPosition from '../pageobjects/test';`

When running my test I receive an error `Error: ReferenceError: window is not defined`.

This might be easy to fix for an experienced programmer.

Help appreciated.

**Comment by christian-bromann** [comment]
@sdoigmm WebdriverIO runs in a Node.js context and not in the browser (unless you use the [browser runner](https://webdriver.io/docs/component-testing)). Therefor you can't use the `window` object, since it doesn't exist in Node. I recommend using the [`execute`](https://webdriver.io/docs/api/browser/execute/) command to run scripts in the browser directly.
